proposals/handler_test.go

```diff
@@ -61,21 +61,22 @@ type testHandler struct {
 func fullMockSet(tb testing.TB) *mockSet {
 	ctrl := gomock.NewController(tb)
 	return &mockSet{
-		mpub: pubsubmock.NewMockPublisher(ctrl),
-		mf:   smocks.NewMockFetcher(ctrl),
-		mbc:  smocks.NewMockBeaconCollector(ctrl),
-		mm:   mocks.NewMockmeshProvider(ctrl),
-		mv:   mocks.NewMockeligibilityValidator(ctrl),
-		md:   mocks.NewMockballotDecoder(ctrl),
-		mvrf: mocks.NewMockvrfVerifier(ctrl),
+		mpub:   pubsubmock.NewMockPublisher(ctrl),
+		mf:     mocks.NewMockFetcher(ctrl),
+		mbc:    mocks.NewMockBeaconCollector(ctrl),
+		mm:     NewMockmeshProvider(ctrl),
+		mv:     NewMockeligibilityValidator(ctrl),
+		md:     NewMockballotDecoder(ctrl),
+		mvrf:   NewMockvrfVerifier(ctrl),
+		mNonce: NewMocknonceFetcher(ctrl),
 	}
 }
 
 func createTestHandler(t *testing.T) *testHandler {
 	types.SetLayersPerEpoch(layersPerEpoch)
 	ms := fullMockSet(t)
 	return &testHandler{
-		Handler: NewHandler(datastore.NewCachedDB(sql.InMemory(), logtest.New(t)), ms.mpub, ms.mf, ms.mbc, ms.mm, ms.md,
+		Handler: NewHandler(datastore.NewCachedDB(sql.InMemory(), logtest.New(t)), ms.mpub, ms.mf, ms.mbc, ms.mm, ms.md, ms.mvrf, ms.mNonce,
```

Comment:
looks like its not needed in this test.

Reply:
Not in this file, but `createTestHandler` is also used in `elegibility_validator_test.go` and there I set expectations on those mocks.